File: 1381. Design a Stack With Increment Operation/solution.cpp
class CustomStack {
private:
    int limit;
    vector<int> stack;
public:
    CustomStack(int maxSize) {
        limit = maxSize;
    }
    
    void push(int x) {
        if (stack.size() < limit) {
            stack.push_back(x);
        }
    }
    
    int pop() {
        if (stack.size() == 0) {
            return -1;
        } else {
            int ret_value = stack.back();
            stack.pop_back();
            return ret_value;
        }
    }
    
    void increment(int k, int val) {
        for (int idx = 0; idx < k and idx < stack.size(); idx += 1) {
            stack[idx] += val;
        }
    }
};

File: 1381. Design a Stack With Increment Operation/solution.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.limit = maxSize
        self.stack = []

    def push(self, x: int) -> None:
        if len(self.stack) < self.limit:
            self.stack.append(x)

    def pop(self) -> int:
        val = -1
        if len(self.stack) != 0:
            val = self.stack[-1]
            del self.stack[-1]
        return val

    def increment(self, k: int, val: int) -> None:
        k = min(k, len(self.stack))
        for idx in range(k):
            self.stack[idx] += val
